node/chainSimulator/components/statusComponents.go

```diff
@@ -87,12 +107,42 @@ func (s *statusComponentsHolder) String() string {
 	return ""
 }
 
-// SetForkDetector will do nothing
-func (s *statusComponentsHolder) SetForkDetector(_ process.ForkDetector) error {
+// SetForkDetector will set the fork detector
+func (s *statusComponentsHolder) SetForkDetector(forkDetector process.ForkDetector) error {
+	if check.IfNil(forkDetector) {
```

Comment:
because of the nature how you use the forkdetector (more than one go routine), you need a mutex here.

Reply:
added